From: Borislav Petkov <bp@suse.de>
Date: Wed Jun 19 10:19:19 CEST 2019
Subject: x86/microcode: Fix microcode hotplug state
References: bsc#1114279
Patch-mainline: never, kabi

Make the new hotplug state alias to an already existing one so that KABI doesn't
complain.

Signed-off-by: Borislav Petkov <bp@suse.de>
--
---
 arch/x86/kernel/cpu/microcode/core.c |    3 +++
 include/linux/cpuhotplug.h           |    3 +--
 2 files changed, 4 insertions(+), 2 deletions(-)

--- a/arch/x86/kernel/cpu/microcode/core.c
+++ b/arch/x86/kernel/cpu/microcode/core.c
@@ -856,6 +856,9 @@ int __init microcode_init(void)
 		goto out_ucode_group;
 
 	register_syscore_ops(&mc_syscore_ops);
+
+#define CPUHP_AP_MICROCODE_LOADER	CPUHP_AP_ARM_MVEBU_COHERENCY
+
 	cpuhp_setup_state_nocalls(CPUHP_AP_MICROCODE_LOADER, "x86/microcode:starting",
 				  mc_cpu_starting, NULL);
 	cpuhp_setup_state_nocalls(CPUHP_AP_ONLINE_DYN, "x86/microcode:online",
--- a/include/linux/cpuhotplug.h
+++ b/include/linux/cpuhotplug.h
@@ -83,8 +83,7 @@ enum cpuhp_state {
 	CPUHP_AP_IRQ_HIP04_STARTING,
 	CPUHP_AP_IRQ_ARMADA_XP_STARTING,
 	CPUHP_AP_IRQ_BCM2836_STARTING,
-	CPUHP_AP_ARM_MVEBU_COHERENCY,
-	CPUHP_AP_MICROCODE_LOADER,
+	CPUHP_AP_ARM_MVEBU_COHERENCY, /* == CPUHP_AP_MICROCODE_LOADER, for KABI */
 	CPUHP_AP_PERF_X86_AMD_UNCORE_STARTING,
 	CPUHP_AP_PERF_X86_STARTING,
 	CPUHP_AP_PERF_X86_AMD_IBS_STARTING,
